src/helpers.js

```diff
@@ -57,10 +60,177 @@ export function idsToIgnoredRules(ruleIds = []) {
+    `Atom version: ${debug.atomVersion}`,
+    `linter-eslint version: ${debug.linterEslintVersion}`,
+    `ESLint version: ${debug.eslintVersion}`,
+    `Hours since last Atom restart: ${debug.hoursSinceRestart}`,
+    `Platform: ${debug.platform}`,
+    `Using ${debug.eslintType} ESLint from ${debug.eslintPath}`,
+    `linter-eslint configuration: ${JSON.stringify(debug.linterEslintConfig, null, 2)}`
+  ]
+  return details.join('\n')
+}
+
+const generateInvalidTrace = async (
+  msgLine, msgCol, msgEndLine, msgEndCol,
+  eslintFullRange, filePath, textEditor, ruleId, message, worker
+) => {
+  let errMsgRange = `${msgLine + 1}:${msgCol}`
+  if (eslintFullRange) {
+    errMsgRange += ` - ${msgEndLine + 1}:${msgEndCol + 1}`
+  }
+  const rangeText = `Requested ${eslintFullRange ? 'start point' : 'range'}: ${errMsgRange}`
+  const issueURL = 'https://github.com/AtomLinter/linter-eslint/issues/new'
+  const titleText = `Invalid position given by '${ruleId}'`
+  const title = encodeURIComponent(titleText)
+  const body = encodeURIComponent([
+    'ESLint returned a point that did not exist in the document being edited.',
+    `Rule: \`${ruleId}\``,
+    rangeText,
+    '', '',
+    '<!-- If at all possible, please include code to reproduce this issue! -->',
+    '', '',
+    'Debug information:',
+    '```json',
+    JSON.stringify(await getDebugInfo(worker), null, 2),
+    '```'
+  ].join('\n'))
+  const newIssueURL = `${issueURL}?title=${title}&body=${body}`
+  return {
+    type: 'Error',
+    severity: 'error',
+    html: `${escapeHTML(titleText)}. See the trace for details. ` +
+      `<a href="${newIssueURL}">Report this!</a>`,
+    filePath,
+    range: rangeFromLineNumber(textEditor, 0),
+    trace: [
+      {
+        type: 'Trace',
+        text: `Original message: ${ruleId} - ${message}`,
+        filePath,
+        severity: 'info',
+      },
+      {
+        type: 'Trace',
+        text: rangeText,
+        filePath,
+        severity: 'info',
+      },
+    ]
+  }
+}
+
+export async function processMessages(response, textEditor, showRule, worker) {
+  return Promise.all(response.map(async ({
+    message, line, severity, ruleId, column, fix, endLine, endColumn
+  }) => {
+    const filePath = textEditor.getPath()
+    const textBuffer = textEditor.getBuffer()
+    let linterFix = null
+    if (fix) {
+      const fixRange = new Range(
+        textBuffer.positionForCharacterIndex(fix.range[0]),
+        textBuffer.positionForCharacterIndex(fix.range[1])
+      )
+      linterFix = {
+        range: fixRange,
+        newText: fix.text
+      }
+    }
+    const msgLine = line - 1
+    let msgCol
+    let msgEndLine
+    let msgEndCol
+    let eslintFullRange = false
+    if (typeof endColumn !== 'undefined' && typeof endLine !== 'undefined') {
+      eslintFullRange = true
+      // Here we always want the column to be a number
+      msgCol = Math.max(0, column - 1)
+      msgEndLine = endLine - 1
+      msgEndCol = endColumn - 1
```

Comment:
It might be good to include a comment about why we are subtracting 1 from a lot of these values, as it isn't immediately obvious.